packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -485,3 +488,85 @@ export const WithLabelInternally = () => {
     />
   )
 }
+
+export const AsyncFetch: StoryObj<SelectPanelPropsAndCustomArgs> = {
+  render: ({initialLoadingType, height, componentManagesLoading}: SelectPanelPropsAndCustomArgs) => {
+    const [loading, setLoading] = React.useState<boolean>(true)
+    const [selected, setSelected] = React.useState<ItemInput[]>([])
+    const [filteredItems, setFilteredItems] = React.useState<ItemInput[]>([])
+    const [open, setOpen] = useState(false)
+    const filterTimerId = useRef<number | null>(null)
+    const {safeSetTimeout, safeClearTimeout} = useSafeTimeout()
+
+    const fetchItems = (query: string) => {
+      setLoading(true)
+
+      if (filterTimerId.current) {
+        safeClearTimeout(filterTimerId.current)
+      }
+
+      filterTimerId.current = safeSetTimeout(() => {
+        setFilteredItems(items.filter(item => item.text.toLowerCase().startsWith(query.toLowerCase())))
+        setLoading(false)
+      }, 2000) as unknown as number
+    }
+
+    const onOpenChange = (value: boolean) => {
+      setLoading(true)
+      setOpen(value)
+      fetchItems('')
+    }
+
+    const loadingProps = {
+      initialLoadingType,
+      ...(componentManagesLoading ? {} : {loading}),
+    }
+
+    return (
+      <SelectPanel
+        title="Select labels"
+        subtitle="Use labels to organize issues and pull requests"
+        renderAnchor={({children, 'aria-labelledby': ariaLabelledBy, ...anchorProps}) => (
+          <Button
+            trailingAction={TriangleDownIcon}
+            aria-labelledby={` ${ariaLabelledBy}`}
+            {...anchorProps}
+            aria-haspopup="dialog"
+          >
+            {children ?? 'Select Labels'}
+          </Button>
+        )}
+        placeholderText="Filter labels"
+        open={open}
+        onOpenChange={onOpenChange}
+        items={filteredItems}
+        selected={selected}
+        onSelectedChange={setSelected}
+        onFilterChange={fetchItems}
+        showItemDividers={true}
+        height={height}
+        {...loadingProps}
+      />
+    )
+  },
+  args: {
+    initialLoadingType: 'spinner',
+    height: 'medium',
+    componentManagesLoading: true,
+  },
+  argTypes: {
+    initialLoadingType: {
+      control: 'select',
+      options: ['spinner', 'skeleton'],
+    },
+    height: {
+      control: 'select',
+      options: ['auto', 'xsmall', 'small', 'medium', 'large', 'xlarge'],
+    },
+    componentManagesLoading: {
+      control: {
+        type: 'boolean',
+      },
```

Comment:
What does this control do? I thought it'd let me turn of the loading in the storybook but found I was seeing the spinners either way

Reply:
This controls whether the component, i.e. `SelectPanel` manages the loading state or if it's controlled outside the component by the consumer. The spinner/skeleton will still show up in both cases, but that's because the consumer, i.e. the storybook story, is passing `loading={true/false}`. You'll notice that, in the case `componentManagesLoading` is `false`, the inline text field spinner appears immediately (indicating the consumer flipped `loading` to `true`), whereas when `componentManagesLoading` is `true`, the text field spinner appears after a short delay.

Reply:
ahh, that makes sense, got it!